scripts.js

```diff
@@ -9,7 +9,12 @@ const colors = [
   'orange'
 ];
 
-const allColors = [...colors, ...colors];
+const totalArrayLength = 10;
+
+const allColors = [
+  ...colors.slice(0, eval(totalArrayLength / 2)),
```

Comment:
This is a brilliant way! Most people would involve a loop.
A few things though. As a general note, try to avoid `eval()` as much as possible. In this case, you don't need it. You only need `eval()` to evaluate code that is not 100% javascript. for example, if the `/` would be dynamical and coming from the DOM, in other words, if it would be a string stored in a variable.
```js
const operator = "/";
const length = 10;
const result = eval(`${length}${operator}2`);
```
in this case, ☝️you will need to use `eval()`.
```js
const length = 10;
const result = length / 2;
```
in this case, ☝️you don't as all the parts of your equation are actually pure JavaScript.

That said, you can also avoid doing this calculation twice. I'm not gonna go into dept how, as I am pretty confident (no pressure 😅) you will be able to figure it out 😉

Reply:
`eval()` used to be considered as **evil** 😅 I never really agreed with it as sometimes you need it, and if it was that **evil**, they would have removed it by now. 
I just found an article called **`eval()` isn’t evil, just misunderstood**.
https://humanwhocodes.com/blog/2013/06/25/eval-isnt-evil-just-misunderstood/